src/main/webapp/components/result-forms/route.js

```diff
@@ -51,8 +58,23 @@ const AddItem = props => {
 }
 
 const Item = props => {
-  const { form, Editor, onDelete } = props
-
+  const { form, Editor, onDelete, user } = props
+
+  const securityAttributes = getSecurityAttributesFromMetacard(form)
+  const canShare = isAdmin(user.email, securityAttributes, form.owner)
+  const canWrite = isWritable(
+    user.email,
+    user.roles,
+    securityAttributes,
+    canShare
+  )
+  const readOnly = isReadOnly(
+    canWrite,
+    canShare,
+    securityAttributes,
+    user.email,
+    user.roles
+  )
```

Comment:
Have you considered using one method that takes in a metacard and returns all three of the properties? I'm thinking of a function call like `const { canShare, canWrite, readOnly } = getSecurityPropertiesFromMetacard(form)`